python/tests/test_logistic_regression.py

```diff
@@ -169,62 +242,74 @@ def test_compat(
             (X, weight.reshape(num_rows, 1), y.reshape(num_rows, 1)), axis=1
         )
 
-        df = spark.createDataFrame(
+        bdf = spark.createDataFrame(
             np_array.tolist(),
             ",".join(schema),
         )
 
-        df = df.withColumn("features", array_to_vector(array(*feature_cols))).drop(
+        bdf = bdf.withColumn("features", array_to_vector(array(*feature_cols))).drop(
             *feature_cols
         )
 
-        lr = _LogisticRegression()
-        # lr = _LogisticRegression(regParam=0.1, standardization=False)
-        # assert lr.getRegParam() == 0.1
-
-        lr.setFeaturesCol("features")
-        # lr.setMaxIter(30)
-        # lr.setRegParam(0.01)
-        lr.setLabelCol("label")
-        if isinstance(lr, SparkLogisticRegression):
-            lr.setWeightCol("weight")
-
-        assert lr.getFeaturesCol() == "features"
-        # assert lr.getMaxIter() == 30
-        # assert lr.getRegParam() == 0.01
-        assert lr.getLabelCol() == "label"
-
-        model = lr.fit(df)
-        coefficients = model.coefficients.toArray()
-        intercept = model.intercept
-
-        if isinstance(lr, SparkLogisticRegression):
-            assert array_equal(coefficients, [-17.65543489, 17.65543489])
-            assert intercept == pytest.approx(0, abs=1e-6)
+        blor = _LogisticRegression(regParam=0.1)
+        assert blor.getRegParam() == 0.1
+
+        blor.setFeaturesCol("features")
+        blor.setMaxIter(10)
+        blor.setRegParam(0.01)
+        blor.setLabelCol("label")
+        if isinstance(blor, SparkLogisticRegression):
+            blor.setWeightCol("weight")
+
+        assert blor.getFeaturesCol() == "features"
+        assert blor.getMaxIter() == 10
+        assert blor.getRegParam() == 0.01
+        assert blor.getLabelCol() == "label"
+
+        blor.clear(blor.maxIter)
+        assert blor.getMaxIter() == 100
+
+        blor_model = blor.fit(bdf)
+
+        blor_model.setFeaturesCol("features")
+        blor_model.setProbabilityCol("newProbability")
```

Comment:
There should be a test to check that probability column has been added during transform and looks reasonable.

Reply:
Will address this in the next PR.